lib/cache.js

```diff
@@ -3,23 +3,22 @@ var fs = require('fs'),
 
 // Loads up all default markers at require time.
 module.exports = [
-    ['base', 'base'],
-    ['mask', 'mask'],
-    ['alphanum', 'symbol']
+    ['markers-src/base', 'base'],
+    ['markers-src/mask', 'mask'],
+    ['markers-src/alphanum', 'symbol'],
+    ['renders', 'symbol']
 ].reduce(function(memo, destsrc) {
-    var dest = destsrc[1],
-        src = destsrc[0],
-        basepath = path.resolve(__dirname + '/../markers-src/' + src);
+    const [src, dest] = destsrc;
+    const basePath = path.resolve(__dirname, '..', src);
 
-    memo[dest] = fs.readdirSync(basepath)
-        .sort()
```

Comment:
Does it hurt to keep this `sort()`? It might make a difference:
* Debugging. It's just nice when we know the elements in the `memo` lookup are in alphabetical order.
* The JavaScript JIT might be able to do some more performance magic when the elements have some stable order.

Reply:
It probably does hurt to keep the sort, because memory is copied around a number of times, some type of sort is executed, and I believe it all has no benefit.  JS objects are unordered and this code is building an object with string keys.  The insertion order makes no difference because memory isn't rearranged inside the object, and the cache will be queried by key, which is hashed to give linear access time.  By the same logic, we could sort the keys by hashed value and insert that way (still with no benefit).

Reply:
Objects being "unordered" is more a myth. I remember some Java version where this was really true. But in the languages we use here (JavaScript as well as PHP) the order of properties in an object is stable.

Anyway. I also think the sort doesn't make a difference. Still wanted to ask the question.

Performance shouldn't be a problem. We know the maximum is about 200 elements.

Reply:
The language [spec](https://262.ecma-international.org/6.0/#sec-ordinary-object-internal-methods-and-internal-slots-ownpropertykeys) does define a default key iteration behavior: first numeric keys are returned in ascending numeric order, then string keys are returned in the order inserted, finally any symbols in the order of insertion.

However, [they say](https://stackoverflow.com/a/30919039) that methods like Object.keys() *don't* reliably follow this implicit order, so it seems that we shouldn't make any assumptions about the order mattering.  For debugging, just run sort() at display time.

There are 1200 icon files in the symbol directory and 750 in the alphanumeric directory.  The worst-case complexity of quicksort is O(N^2), so I would keep the performance question on the table.